### PR TITLE
PR details modal

### PR DESCRIPTION
**Pull request by estib-vega**
## ☕️ Reasoning
We should give the user the ability to see and edit the PR details without having to go out of the app.

This pull request introduces features such as a preview of the PR title and body, the ability to edit PR contents, and keyboard shortcuts, users will have a more intuitive and streamlined process for managing pull requests.

## 🧢 Changes

- Added a button on the Pull Request card, displaying a modal with PR details (for existing PRs).
- Implemented a preview modal that displays the title and body of the PR during the creation process for user edition and confirmation.
- The PR body will by default be populated with the specified PR template and or if it's only one commit, with that commit's title and body (as usual).
- ✨Incorporated functionality to generate the PR description automatically from commit messages, with an optional user prompt available.
- Implemented keyboard input capturing within the modal for a smoother user interaction experience:
    - Command or Ctrl + e toggles edit mode
    - Command or Ctrl + g generates a PR description
    - Command or Ctrl + Enter creates the PR

This comprehensive set of updates aims to elevate the overall experience of managing pull requests within the application.

## Visuals

https://github.com/user-attachments/assets/4ef4cdfd-6273-43cc-a248-20ce3fce4f2d

### The title of the PR card now is a button that opens the read-only version of the modal

Screenshot 2024-10-04 at 10 10 22

### CODE REVIEW

**Comment by estib-vega** [comment]
### Some more info about the internals

The generation of the PR can be done even before entering the 'edit' view, so that you can start the PR, click `cmd` + `g`, wait for the generation and then hit `cmd` + `Enter` and create the PR.

The description generation uses the following things as context:
- PR current title
- PR current body contents
- PR template (if set)
- The commit messages in full and in chronological order

The user can give a prompt to tune their PR as they want from the edit mode.
This allows users to generate an initial description and then work on top of it by updating the prompt or the title/body contents manually.

**Comment by PavelLaptev** [comment]
Thanks, @estib-vega! That's a great addition.
I'll need to design it in Figma now. This is also why I'm asking to be ping me about any UI changes, so I can sync everything in Figma, ensuring we're aligned on the app's look.

**Comment by estib-vega** [comment]
> Thanks, @estib-vega! That's a great addition. I'll need to design it in Figma now. This is also why I'm asking to be ping me about any UI changes, so I can sync everything in Figma, ensuring we're aligned on the app's look and feel.

@PavelLaptev Makes sense 👍 
Take your time :)

File: packages/ui/src/lib/Modal.svelte [hunk 1]
```diff
@@ -129,16 +135,19 @@
...
+	:global(dialog[open]::backdrop) {
+		background-color: rgba(214, 214, 214, 0.4);
 		animation: dialog-fade 0.15s ease-in;
 	}
 
-	html.dark dialog[open]::backdrop {
-		/* NOTE: temporarily hardcoded dark var(--clr-overlay-bg); */
-		background-color: color(srgb 0.8392156862745098 0.8392156862745098 0.8392156862745098 / 0.4);
+	/* backdrop dark */
+	/* NOTE: temporarily hardcoded dark var(--clr-overlay-bg); */
```
Comment (mtsgrd) [inline packages/ui/src/lib/Modal.svelte:152]: How come these are a) temporary, b) hardcoded?

Comment (estib-vega) [reply packages/ui/src/lib/Modal.svelte:152]: cc @PavelLaptev :)

**Comment by estib-vega** [comment]
@PavelLaptev 
Some updates for the Stack Series PR button

Screenshot 2024-10-04 at 14 34 28

File: packages/ui/src/lib/Modal.svelte (hunk 1 above)
Comment (PavelLaptev) [reply packages/ui/src/lib/Modal.svelte:152]: @mtsgrd there was an issue with this `backdrop` thing. it doesn't work with the native `dialog` component. So we're waiting when this bug will be fixed. And this thing doesn't work only on Linux machines by some reason.

**Comment by PavelLaptev** [comment]
Update for the "Enable PR templates" section:

image

Also updated in [Figma](https://www.figma.com/design/FbeLt0yjY9RiNn8MXUXsYs/Client-design?node-id=6798-46579&t=BHma0Dyzvv9FlRvx-1)

**Review by Caleb-T-Owens — APPROVED** [review]
LGTM

**Comment by estib-vega** [comment]
FYI: don't merge this yet, waiting on design sync

**Review by PavelLaptev — APPROVED** [review]

**Pull request merged**